proof_ipa.go

```diff
@@ -370,3 +442,36 @@ func TreeFromProof(proof *Proof, rootC *Point) (VerkleNode, error) {
 
 	return root, nil
 }
+
+// PostStateTreeFromProof uses the pre-state trie and the list of updated values
+// to produce the stateless post-state trie.
+func PostStateTreeFromProof(preroot VerkleNode, statediff StateDiff) (VerkleNode, error) {
+	postroot := preroot.Copy()
+
+	for _, stemstatediff := range statediff {
+		var (
+			values     = make([][]byte, NodeWidth)
+			overwrites bool
+		)
+
+		for _, suffixdiff := range stemstatediff.SuffixDiffs {
+			if len(suffixdiff.NewValue) > 0 {
+				// If the prestate is present, it means that at least one
+				// post value will be non-nil. In this case, make a copy
+				// of the pre tree and update all post values.
```

Comment:
I find this comment a bit confusing. Nothing about this `for` loop references the pre-state (i.e: `make a copy` or similar)